tests/fixtures/fixtures.cpp

```diff
@@ -0,0 +1,115 @@
+
+  // get the directory of the test executable
+  testBinaryDir = std::filesystem::path(testBinary).parent_path();
+
+  // If testBinaryDir is empty or `.` then set it to the current directory
+  // maybe some better options here: https://stackoverflow.com/questions/875249/how-to-get-current-directory
+  if (testBinaryDir.empty() || testBinaryDir.string() == ".") {
+    testBinaryDir = std::filesystem::current_path();
+  }
+
+  sbuf = std::cout.rdbuf();  // save cout buffer (std::cout)
+  std::cout.rdbuf(cout_buffer.rdbuf());  // redirect cout to buffer (std::cout)
+
+  // Default to the verbose level in case some test fails
+  display_device::logger_t::get().set_log_level(display_device::logger_t::log_level_e::verbose);
+}
+
+void
+BaseTest::TearDown() {
+  display_device::logger_t::get().set_custom_callback(nullptr);  // restore the default callback to avoid potential leaks
+  std::cout.rdbuf(sbuf);  // restore cout buffer
+
+  // get test info
+  const ::testing::TestInfo *const test_info = ::testing::UnitTest::GetInstance()->current_test_info();
+
+  if (test_info->result()->Failed()) {
+    std::cout << std::endl
+              << "Test failed: " << test_info->name() << std::endl
+              << std::endl
+              << "Captured cout:" << std::endl
+              << cout_buffer.str() << std::endl
+              << "Captured stdout:" << std::endl
+              << stdout_buffer.str() << std::endl
+              << "Captured stderr:" << std::endl
+              << stderr_buffer.str() << std::endl;
+  }
+
+  sbuf = nullptr;  // clear sbuf
+  if (pipe_stdout) {
+    pclose(pipe_stdout);
+    pipe_stdout = nullptr;
+  }
+  if (pipe_stderr) {
+    pclose(pipe_stderr);
+    pipe_stderr = nullptr;
+  }
+}
+
+int
+BaseTest::exec(const char *cmd) {
+  std::array<char, 128> buffer {};
+  pipe_stdout = popen((std::string(cmd) + " 2>&1").c_str(), "r");
+  pipe_stderr = popen((std::string(cmd) + " 2>&1").c_str(), "r");
+  if (!pipe_stdout || !pipe_stderr) {
+    throw std::runtime_error("popen() failed!");
+  }
+  while (fgets(buffer.data(), buffer.size(), pipe_stdout) != nullptr) {
+    stdout_buffer << buffer.data();
+  }
+  while (fgets(buffer.data(), buffer.size(), pipe_stderr) != nullptr) {
+    stderr_buffer << buffer.data();
+  }
+  int returnCode = pclose(pipe_stdout);
+  pipe_stdout = nullptr;
+  if (returnCode != 0) {
+    std::cout << "Error: " << stderr_buffer.str() << std::endl
+              << "Return code: " << returnCode << std::endl;
+  }
+  return returnCode;
+}
+
+void
+LinuxTest::SetUp() {
+#ifndef __linux__
+  GTEST_SKIP_("Skipping, this test is for Linux only.");
+#endif
+  BaseTest::SetUp();
+}
+
+void
+MacOSTest::SetUp() {
+#if !defined(__APPLE__) || !defined(__MACH__)
+  GTEST_SKIP_("Skipping, this test is for macOS only.");
+#endif
+  BaseTest::SetUp();
+}
+
+void
+WindowsTest::SetUp() {
+#ifndef _WIN32
+  GTEST_SKIP_("Skipping, this test is for Windows only.");
+#endif
+  BaseTest::SetUp();
+}
```

Comment:
I believe each of these needs a TearDown, otherwise the BaseTest::TearDown() is not called.

Reply:
Why would it not be called? We are not overriding `TearDown` for `LinuxTest` (for example) so the `BaseTest::TearDown` is used automatically.

Reply:
I don't know, but in my previous tests it was not called. You can test it by creating a test with some cout, that is forced to fail.

Reply:
Hmm, it's called for me:
![image](https://github.com/LizardByte/libdisplaydevice/assets/22381748/52e96d77-816a-45fb-891d-d4068830a96d)